### ISSUE TITLE
Add "Device" column to DownloadLog.csv

### ISSUE THREAD

**Issue by scienmind** [canton7/SyncTrayzor#332]
Since "Global Changes" (a short list of recent changes) was implemented in the core ST (with 0.14.17,  think...) the source device name that made the file change is also logged. 

So it can be nice to add this to the saved log as well :)
smthn like this: `"Date,Device,Folder,Path,Type,Action,Error"`

**Comment by canton7** [comment]
That's only available by polling, right? It's not available on the events api?

**Comment by canton7** [comment]
Also isn't the global changes purely about who changes files, rather than actual downloads? 

Having a "device" against the downloads wouldn't make sense surely, as files aren't downloaded from a single device?

**Comment by scienmind** [comment]
> That's only available by polling, right? It's not available on the events api?

Not sure... it was introduced in this [PR](https://github.com/syncthing/syncthing/pull/3694), with some fixes later. 
Although, from this forum [thread](https://forum.syncthing.net/t/global-changes-in-syncthing/9039), it seems like it is available via API.

> Also isn't the global changes purely about who changes files, rather than actual downloads?

Hmm, you are correct... the concepts are a bit different. 
It's not an issue though, I think It would be a nice enhancement. We should just list it as "Modification Origin" (or something in this direction) instead of simply "Device". Less ambiguous and more informative.

Another aspect - this will also give us a listing of local modifications sent to other devices (which aren't logged now, as they are not "downloads"). 
From my perspective at least, it is a desired behavior. Add a column "Sent/Received, and maybe change filename to something more suitable, like "FileModificationLog.csv" 
(all this is more or less what I had in mind making the original feature request for this log... )

**Comment by canton7** [comment]
> Although, from this forum thread, it seems like it is available via API.

Yes, but via the REST API, rather than the events API - which means that if we're constantly monitoring it will add to Syncthing's CPU (and possibly memory) usage, even when it's idle.

>  We should just list it as "Modification Origin" (or something in this direction) instead of simply "Device". Less ambiguous and more informative.

This requires stitching together two different data sources. Not impossible by any means, but much more complex than simply adding another column.

> Another aspect - this will also give us a listing of local modifications sent to other devices (which aren't logged now, as they are not "downloads").

Local modifications? Yes. Sent to other devices? No, there's no indication of that. 

> Add a column "Sent/Received, and maybe change filename to something more suitable, like "FileModificationLog.csv"

Again this doesn't track what is sent to other devices. If all of your other devices are offline, it will still appear as "sent".

**Comment by scienmind** [comment]
I see....

Sounds like a lot of work beyond "adding columns and renaming stuff".

Given the (probably) low interest in this functionality, I guess we can leave the download log file as it is for now (or even close this). 
Until either someone submits a PR. or until we accidentally discover that the key to the world peace lies within a properly structured, detailed file transaction log functionality... :dove:
